### PR TITLE
feat(search): add validation props (FE-5292)

### PR DESCRIPTION
**Pull request by grabkowski**
### Proposed behaviour

Add validation props to the Search component and pass them down to the Textbox

### Current behaviour

Validations could not be applied to the Search component

### Checklist

- [x] Commits follow our style guide
- [x] Related issues linked in commit messages if required
~~- [ ] Screenshots are included in the PR if useful~~
~~- [ ] All themes are supported if required~~
~~- [ ] Unit tests added or updated if required~~
~~- [ ] Cypress automation tests added or updated if required~~
- [x] Storybook added or updated if required
~~- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required~~
- [x] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

### CODE REVIEW

File: src/components/search/search.stories.mdx [hunk 1]
```diff
@@ -177,6 +177,101 @@ In this example the `variant` prop is "dark" and no search button is present.
...
+<Canvas>
+  <Story name="validations - string - component">
+    {() => {
+      const [state, setState] = useState({
+        error: "<>/",
+        warning: "<>/",
+        info: "<>/",
+      });
+      const handleChange = (validation) => (e) => {
+        setState({ ...state, [validation]: e.target.value.rawValue });
```
Comment (robinzigmond) [inline src/components/search/search.stories.mdx:201]:
```suggestion
        setState({ ...state, [validation]: e.target.value });
```

Not sure what the `rawValue` is doing here - the component is crashing for me with this in, and `e.target.value` is just the string value. (And this matches the "controlled" story above.)

File: src/components/search/search.stories.mdx [hunk 2]
```diff
@@ -177,6 +177,101 @@ In this example the `variant` prop is "dark" and no search button is present.
...
+
+For more information check our [Validations](?path=/docs/documentation-validations--page "Validations") documentation page
+
+#### As a string
+
+<Canvas>
+  <Story name="validations - string - component">
+    {() => {
+      const [state, setState] = useState({
+        error: "<>/",
```
Comment (robinzigmond) [inline src/components/search/search.stories.mdx:196]: I know you want to show something with "invalid characters" to make it look realistic, but this looks weird to me with those values. I'd put some plain text in there, maybe with one invalid character like `foo/`? Up to you, I don't mind leaving this as it is if you prefer it.

File: src/components/search/search.stories.mdx [hunk 3]
```diff
@@ -177,6 +177,101 @@ In this example the `variant` prop is "dark" and no search button is present.
...
+It is possible to use the `tooltipPosition` to override the default placement of tooltips rendered as part of this component.
+
+<Canvas>
+  <Story
+    name="validations - string - with tooltipPosition overriden - component"
+    parameters={{ chromatic: { disable: true } }}
+  >
+    {() => {
+      const [state, setState] = useState({
+        error: "<>/",
```
Comment (robinzigmond) [inline src/components/search/search.stories.mdx:226]: same comment as above re these initial values

File: src/components/search/search.stories.mdx [hunk 4]
```diff
@@ -177,6 +177,101 @@ In this example the `variant` prop is "dark" and no search button is present.
...
+        warning: "<>/",
+        info: "<>/",
+      });
+      const handleChange = (validation) => (e) => {
+        setState({ ...state, [validation]: e.target.value.rawValue });
```
Comment (robinzigmond) [inline src/components/search/search.stories.mdx:231]:
```suggestion
        setState({ ...state, [validation]: e.target.value });
```

File: src/components/search/search.stories.mdx [hunk 5]
```diff
@@ -177,6 +177,101 @@ In this example the `variant` prop is "dark" and no search button is present.
...
+  </Story>
+</Canvas>
+
+#### As a boolean
+
+<Canvas>
+  <Story name="validations - boolean">
+    {() => {
+      const [state, setState] = useState({
+        error: "<>/",
```
Comment (robinzigmond) [inline src/components/search/search.stories.mdx:254]: same comment as above re these initial values

File: src/components/search/search.stories.mdx [hunk 6]
```diff
@@ -177,6 +177,101 @@ In this example the `variant` prop is "dark" and no search button is present.
...
+        warning: "<>/",
+        info: "<>/",
+      });
+      const handleChange = (validation) => (e) => {
+        setState({ ...state, [validation]: e.target.value.rawValue });
```
Comment (robinzigmond) [inline src/components/search/search.stories.mdx:259]:
```suggestion
        setState({ ...state, [validation]: e.target.value });
```

**Review by robinzigmond — CHANGES_REQUESTED** [review]
I think we should add some unit tests for these validation props on Search. I know it will already be covered by the tests on Textbox where we're passing them through to but it would be good to make sure they're getting passed through to make sure we don't accidentally remove these from Search.

**Comment by grabkowski** [comment]
> I think we should add some unit tests for these validation props on Search. I know it will already be covered by the tests on Textbox where we're passing them through to but it would be good to make sure they're getting passed through to make sure we don't accidentally remove these from Search.

Could you tell me why do we need unit tests for that, when we have everything already covered by a Chromatic snapshot?

File: src/components/search/search.stories.mdx (hunk 1 above)
Comment (grabkowski) [reply src/components/search/search.stories.mdx]: my mistake, I'll remove that

**Comment by robinzigmond** [comment]
> > I think we should add some unit tests for these validation props on Search. I know it will already be covered by the tests on Textbox where we're passing them through to but it would be good to make sure they're getting passed through to make sure we don't accidentally remove these from Search.
> 
> Could you tell me why do we need unit tests for that, when we have everything already covered by a Chromatic snapshot?

Good point, I'd forgotten about Chromatic. I'll let this one go :)

**Review by robinzigmond — APPROVED** [review]

**Review by mkrds — APPROVED** [review]

**Review by robinzigmond — APPROVED** [review]

**Review by mkrds — APPROVED** [review]

**Pull request merged**